### PR TITLE
chain: add standard interface, run, call and llm 

### PR DESCRIPTION
**Pull request by FluffyKebab**
- Add interface for chains outside of exp. Maybe the call method in the interface should be renamed to avoid confiusion with the call function that should be used by users. 
 - Add run and call functions.
https://github.com/hwchase17/langchainjs/blob/main/langchain/src/chains/base.ts
https://github.com/hwchase17/langchain/blob/master/langchain/chains/base.py
 - Add llm chain.
 https://github.com/hwchase17/langchainjs/blob/main/langchain/src/chains/llm_chain.ts
https://github.com/hwchase17/langchain/blob/master/langchain/chains/llm.py

### PR Checklist

- [x] Read the [Contributing documentation](https://github.com/tmc/langchaingo/blob/main/CONTRIBUTING.md).
- [x] Read the [Code of conduct documentation](https://github.com/tmc/langchaingo/blob/main/CODE_OF_CONDUCT.md).
- [x] Name your Pull Request title clearly, concisely, and prefixed with the name of primarily affected package you changed according to [Good commit messages](https://go.dev/doc/contribute#commit_messages) (such as `memory: add interfaces for X, Y` or `util: add whizzbang helpers`).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Describes source of new concepts.
- [x] References existing implementations as appropriate.
- [x] Contains test coverage for new functions.
- [x] Passes all [`golangci-lint`](https://golangci-lint.run/) checks.
- [x] I am an existing contributor.

### CODE REVIEW

File: chain/chain.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
+package chain
```
Comment (tmc) [inline chain/chain.go:1]: this package should be "chains" to reflect upstream/python

File: chain/chain.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	ErrMultipleOutputsInRun = errors.New("run not supported in chain with more then one expected output")
+	// ErrMultipleOutputsInRun is returned in the run function if the chain returns
+	// a value that is not a string.
+	ErrWrongOutputTypeInRun = errors.New("run not supported in chain that returns value that is not string")
+)
+
+// Cain is the interface all chains must implement.
+type Chain interface {
+	// Call runs the logic of the chain and returns the output. This method should
+	// not be called directly. Use rather the Call function that handles the memory
```
Comment (tmc) [inline chain/chain.go:32]: hmm yeah this is a potentially confusing API, I'll put some thought into this.

File: chain/chain.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	ErrMultipleInputsInRun = errors.New("run not supported in chain with more then one expected input")
+	// ErrMultipleOutputsInRun is returned in the run function if the chain expects
+	// more then one output values.
+	ErrMultipleOutputsInRun = errors.New("run not supported in chain with more then one expected output")
+	// ErrMultipleOutputsInRun is returned in the run function if the chain returns
+	// a value that is not a string.
+	ErrWrongOutputTypeInRun = errors.New("run not supported in chain that returns value that is not string")
+)
+
+// Cain is the interface all chains must implement.
```
Comment (abhinavmaury) [inline chain/chain.go:29]: a typo here

**Review by tmc — APPROVED** [review]
I'll think about the documentation and how/if we can clear this up in terms of naming but I'd prefer this get in sooner than later. LGTM.

**Pull request merged**